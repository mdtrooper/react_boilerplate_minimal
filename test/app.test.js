import React from 'react';
import App from '../src/app';
import renderer from 'react-test-renderer';

test('App changes the class when hovered', () => {
  const component = renderer.create(
    <App text="Hello world" />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
