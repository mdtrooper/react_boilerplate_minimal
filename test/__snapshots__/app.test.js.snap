// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App changes the class when hovered 1`] = `
<h1
  className="app"
>
  Hello world
</h1>
`;
